### PR TITLE
Use the Dribble ShotsRepository in search also

### LINKED ISSUE #481: Search result crashes
When trying to perform a search and pressing on a result, I get this crash:
> 2018-08-21 12:12:58.411 21463-21463/io.plaidapp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.plaidapp, PID: 21463
    java.lang.RuntimeException: Unable to start activity ComponentInfo{io.plaidapp/io.plaidapp.dribbble.ui.shot.ShotActivity}: java.lang.IllegalStateException: Could not retrieve shot 4635899
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2913)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalStateException: Could not retrieve shot 4635899
        at io.plaidapp.dribbble.ui.shot.ShotViewModel.<init>(ShotViewModel.kt:50)
        at io.plaidapp.dribbble.ui.shot.ShotViewModelFactory.create(ShotViewModelFactory.kt:40)
        at android.arch.lifecycle.ViewModelProvider.get(ViewModelProvider.java:134)
        at android.arch.lifecycle.ViewModelProvider.get(ViewModelProvider.java:102)
        at io.plaidapp.dribbble.ui.shot.ShotActivity.onCreate(ShotActivity.kt:126)
        at android.app.Activity.performCreate(Activity.java:7136)
        at android.app.Activity.performCreate(Activity.java:7127)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893)

### PR DESCRIPTION
**Pull request by florina-muntenescu**
Fixes #481

### CODE REVIEW

File: core/src/main/java/io/plaidapp/core/data/SearchDataManager.java [hunk 1]
```diff
@@ -41,15 +37,16 @@
...
     // state
     private String query = "";
     private int page = 1;
-    private List<Call> inflight;
 
     public SearchDataManager(Context context) {
         super();
         searchStoriesUseCase = Injection.provideSearchStoriesUseCase(context);
-        inflight = new ArrayList<>();
+        shotsRepository = DribbbleInjection.provideShotsRepository();
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/data/SearchDataManager.java]: is this gonna be swapped for Dagger at some point or not a priority?

File: core/src/main/java/io/plaidapp/core/data/SearchDataManager.java [hunk 2]
```diff
@@ -76,13 +73,8 @@ public void clear() {
...
     public void cancelLoading() {
-        if (inflight.size() > 0) {
-            for (Call call : inflight) {
-                call.cancel();
-            }
-            inflight.clear();
-            searchStoriesUseCase.cancelAllRequests();
-        }
+        searchStoriesUseCase.cancelAllRequests();
+        shotsRepository.cancelAllSearches();
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/data/SearchDataManager.java]: do the order of these calls matter? The removed code (`inflight` stuff) would cancel all the calls before cancelling all the requests.

File: core/src/main/java/io/plaidapp/core/data/SearchDataManager.java [hunk 3]
```diff
@@ -97,38 +89,25 @@ private void searchDesignerNews(final String query, final int resultsPage) {
...
-                        onDataLoaded(shots);
-                    }
-                    inflight.remove(dribbbleSearchCall);
-                } else {
-                    failure(dribbbleSearchCall);
+        shotsRepository.search(query, page, result -> {
+            loadFinished();
+            if (result instanceof Result.Success) {
+                List<Shot> shots = ((Result.Success<? extends List<Shot>>) result).getData();
+                if (shots != null) {
```
Comment (ataulm) [inline core/src/main/java/io/plaidapp/core/data/SearchDataManager.java]: Can we avoid the null check? If it's Result.Success, it should have data.

File: core/src/main/java/io/plaidapp/core/data/SearchDataManager.java (hunk 2 above)
Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/data/SearchDataManager.java]: nope.

File: core/src/main/java/io/plaidapp/core/data/SearchDataManager.java (hunk 3 above)
Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/data/SearchDataManager.java]: done!

**Pull request merged**